### ISSUE TITLE
How to expose just get endpoints from Routers

### ISSUE THREAD

**Issue by junoriosity** [tiangolo/fastapi#5620]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#router_todo.py
from fastapi import FastAPI

router = APIRouter()

@router.get("/")
def root():
    return "todo"

@router.post("/todo")
def create_new():
    return "create new item"



#main.py
from fastapi import FastAPI
import router_todo

app = FastAPI()
app.include_router(router_todo.router, prefix="/todo", tags=["todo"])
```

### Description

I have several Routers whose endpoints I expose on a single app. These endpoints can be anything from post, get, put, delete.

This works nicely, but I am wondering whether there is an elegant solution to just expose the get endpoints for routers?

Of course, I could wrap every endpoint definition in an if condition, but perhaps there is a better way to do it. 😀

### Operating System

macOS

### FastAPI Version

0.85.0

### Python Version

3.10

**Comment by yinziyan1206** [comment]
you can also wrap include_router too

**Comment by junoriosity** [comment]
@yinziyan1206 Yes, many thanks for getting back to me.

My concern is as follows: Assume I have some more similar routers. And for each of these I just want to expose get endpoints and this in the most elegant way possible.

**Comment by tiangolo** [comment]
Humm, I had never had to do something like that, I imagine you would have to iterate in each one of the routes and inspect their methods. But you would have to do it by hand.

Probably even better to think and consider, why do you want to do that? What use case are you solving?

**Comment by junoriosity** [comment]
@tiangolo Sure thing, assume I have different regions and a geo-replicated postgresql, where I can read/write in one and just read in all others.

Hence, it would make life much easier if I have the same code basis but can turn off all endpoints but the get endpoints.

**Comment by tiangolo** [comment]
Ah, I would not remove the endpoints but add a dependency that checks that the request comes from the right region.

**Comment by junoriosity** [comment]
@tiangolo That sounds like an elegant solution. 

Could you give me a small code snippet how to do it?

**Comment by tiangolo** [comment]
You can check the docs here: https://fastapi.tiangolo.com/tutorial/dependencies/

**Comment by junoriosity** [comment]
@tiangolo Many thanks for providing me with the link. I just use dependencies for providing authentication or database connection. But how does it work to make an endpoint invisible depending on variable?

**Comment by Abdeldjalil-H** [comment]
@junoriosity 

You can define a dependency like that

```python
allowed_methods = ["GET"]

def check_method(request: Request):
    if request.method not in allowed_methods:
        raise HTTPException(status_code=403)
```

and add it to the router
```python
router = APIRouter(dependencies=[Depends (check_method)])
```

**Comment by tiangolo** [comment]
Yep. The thing is, I personally would consider the idea of making the endpoints invisible in some cases not the right approach.

I think the endpoints would be "visible" but would return an error if called from the wrong place or in the wrong way.

But anyway, if you really wanna make them disappear, you probably do need a bunch of `if` blocks around them.
